libpod/container_api.go

```diff
@@ -833,8 +833,14 @@ func (c *Container) Refresh(ctx context.Context) error {
 // ContainerCheckpointOptions is a struct used to pass the parameters
 // for checkpointing (and restoring) to the corresponding functions
 type ContainerCheckpointOptions struct {
-	Keep        bool
+	// Keep tells the API to not delete checkpoint artifacts
+	Keep bool
+	// KeepRunning tells the API to keep the container running
+	// after writing the checkpoint to disk
 	KeepRunning bool
+	// TCPEstablished tells the API to checkpoint a container
+	// even if it contains established TCP connections
+	TCPEstablished bool
```

Comment:
Should this default to true?

Reply:
This is something I was also thinking about. The main reason to default to false is that if there are established TCP connections the user might want to explicitly tell to actually checkpoint and restore. Maybe the environment does not handle checkpointing and restoring established TCP connections well, because an external firewall rule is needed to forward the traffic to the container and thus the migration of TCP connections makes no sense.

So I am unsure, most of the time I think, yes it should default to true as it automatically takes care of everything, but maybe the user should explicitly ask for it to not end up in a situation where the established connection cannot be forwarded through the port forwarding correctly.

I was following CRIU's and runc's default but for convenience we could default it to true. If the established TCP connections can be migrated, it is a win for the user. If not, the migration should still work and the TCP stack needs to handle the reconnection.

Reply:
if CRIU and runc default to false, I'd vote to keep this one false too.

Reply:
I agree that tcp-established should be set to false by default. A user can always add `tcp-established` to the CRIU's configuration file to change this behaviour.